src/host/screenInfo.cpp

```diff
@@ -1190,20 +1189,6 @@ void SCREEN_INFORMATION::_InternalSetViewportSize(const COORD* const pcoordSize,
                 // Adjust the top of the window instead of the bottom
                 // (so the lines slide upward)
                 srNewViewport.Top -= DeltaY;
-
-                // If we happened to move the top of the window past
-                // the 0th row (first row in the buffer)
-                if (srNewViewport.Top < 0)

```

Comment:
Don't totally understand why this disappeared

Reply:
I removed that code while debugging the crash issue.
Check out my commit here: [`4c1404d` (#12719)](https://github.com/microsoft/terminal/pull/12719/commits/4c1404d524dd1ba437bc681bd3b3095b5fe38f95)

Line 1227-1231 it says:
```cpp
if (srNewViewport.Top < 0)
{
    srNewViewport.Bottom -= srNewViewport.Top;
    srNewViewport.Top = 0;
}
```

That if condition makes the removed code with its fail-fast, etc. redundant.

Reply:
I didn't look at that block you introduced below as being related since the comment said "off the right" instead of "off the bottom"!